tests/test_visitors/test_ast/test_annotations/test_argument_annotations.py

```diff
@@ -68,6 +75,33 @@ def function(empty_arg: Literal[None]): ...
 def function(empty_arg: Union[Literal[None], Optional[int]]): ...
 """
 
+wrong_nested_literal_annotation = """
+def function(arg: Literal[Literal[1, 2], 3]): ...
+"""
+wrong_nested_union_annotation = """
+def function(arg: Union[Union[int, str], float]): ...
+"""
+
+
+@pytest.mark.parametrize('code', [
+    wrong_nested_literal_annotation,
```

Comment:
We also need tests with `typing.Union`, `typing.Literal`, and `typing.Annotated`

Reply:
I'm sorry, could you clarify what `typing.Annotated` means? I can't find it in the python documentation.

Reply:
It is not yet documented. Here's the best resource I know on this topic: https://github.com/python/mypy/search?q=annotated&unscoped_q=annotated

Reply:
This is turning out to be tricker than I thought. Adding logic to support prefixed versions (`typing.Literal`, `typing.Union`, etc) makes flake8 complain of function being too complex, and refactoring it into a separate member function makes flake8 complain about there being too many member functions. Feels like the only option is to refactor the violation into a separate visitor. On a related note, rule forbidding `Literal[None]` will not trigger as well when encountering `typing.Literal[None]`.

Reply:
Separate visitor sound fine! 

P.S. That's exactly why we do have these complexity rules: they force us to have better design (in a way you described).